drivers/sensor/ens210/ens210.c

```diff
@@ -46,7 +46,47 @@ static int ens210_sample_fetch(struct device *dev, enum sensor_channel chan)
 	u32_t temp_valid, humidity_valid;
 #endif /* CONFIG_ENS210_CRC_CHECK */
 
-	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+#if defined(CONFIG_ENS210_TEMPERATURE_SINGLE) \
+	|| defined(CONFIG_ENS210_HUMIDITY_SINGLE)
+
+	const struct ens210_sens_start sense_start = {
+		.t_start = ENS210_T_START && (chan == SENSOR_CHAN_ALL
+				|| chan == SENSOR_CHAN_AMBIENT_TEMP),
+		.h_start = ENS210_H_START && (chan == SENSOR_CHAN_ALL
+				|| chan == SENSOR_CHAN_HUMIDITY)
+	};
+
+	u8_t buf;
+#endif
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL
+			|| chan == SENSOR_CHAN_AMBIENT_TEMP
+			|| chan == SENSOR_CHAN_HUMIDITY);
+
+#if defined(CONFIG_ENS210_TEMPERATURE_SINGLE) \
+	|| defined(CONFIG_ENS210_HUMIDITY_SINGLE)
+	/* Start measuring */
+	ret = i2c_reg_write_byte(drv_data->i2c,
+			DT_INST_0_AMS_ENS210_BASE_ADDRESS,
+			ENS210_REG_SENS_START, *(u8_t *)&sense_start);
+	if (ret < 0) {
+		LOG_ERR("Failed to set SENS_START to 0x%x",
+				*(u8_t *)&sense_start);
+		return -EIO;
+	}
+
+	do {
+		k_sleep(K_MSEC(2));
+		ret =  i2c_reg_read_byte(drv_data->i2c,
+				DT_INST_0_AMS_ENS210_BASE_ADDRESS,
+				ENS210_REG_SENS_START, &buf);
+
+		if (ret < 0) {
+			LOG_ERR("Failed to read SENS_STAT");
+		}
+	} while (buf & *(u8_t *)&sense_start);
```

Comment:
Just a proposal, but can you move this part to a static function? Would be cleaner.